domain/application/service/status.go

```diff
@@ -150,13 +150,13 @@ func encodeUnitAgentStatus(s *status.StatusInfo) (*application.StatusInfo[applic
 	}, nil
 }
 
-// encodeUnitWorkloadStatus converts a core status info to a db status info.
-func encodeUnitWorkloadStatus(s *status.StatusInfo) (*application.StatusInfo[application.UnitWorkloadStatusType], error) {
+// encodeWorkloadStatus converts a core status info to a db status info.
+func encodeWorkloadStatus(s *status.StatusInfo) (*application.StatusInfo[application.WorkloadStatusType], error) {
 	if s == nil {
 		return nil, nil
 	}
 
-	encodedStatus, err := encodeUnitWorkloadStatusType(s.Status)
+	status, err := encodeWorkloadStatusType(s.Status)
```

Comment:
I renamed these to `encodedStatus` to stop this variable shadowing the `status` package.

I don't mind changing them back, but you should do so everywhere if you do, and maybe alias the `status` package to `corestatus`

Reply:
I made the change before yours, I can put it back to encodedStatus.